shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -291,15 +287,10 @@ object FileFormatWriter extends Logging {
             empty2NullPlan,
             Some(ConcurrentOutputWriterSpec(maxWriters, () => sortPlan.createSorter())))
         } else {
-          if (staticPartitionWriteOnly && nativeEnabled) {
```

Comment:
There is no need to add the sort operator before the static partition write. Can we still remove the sort operator for the static partition write?

Reply:
I debugged on this. And it seems we have `orderingMatched = true` in the case that all the partitions are static. So this code path will not be invoked. Would you help me confirm that?